iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Network.java

```diff
@@ -859,4 +860,48 @@ default int getConnectableCount() {
     default IdentifiableType getType() {
         return IdentifiableType.NETWORK;
     }
+
+    /**
+     * If network is valid, do nothing.<br>
+     * If network not valid, check if each network component is valid. A {@link ValidationException} is thrown with an explicit message if one network component is not valid.<br>
+     * If all network components are valid, network validation status is updated to true.
+     * Return the network validation status.
+     */
+    default ValidationLevel runValidationChecks() {
+        return runValidationChecks(true);
+    }
+
+    /**
+     * If network is valid, do nothing.<br>
+     * If network not valid and <code>throwsException</code> is <code>true</code>, check if each network component is valid. A {@link ValidationException} is thrown with an explicit message if one network component is not valid.<br>
+     * If all network components are valid, network validation status is updated to true.
+     * Return the network validation status.
+     */
+    default ValidationLevel runValidationChecks(boolean throwsException) {
+        return runValidationChecks(throwsException, Reporter.NO_OP);
+    }
+
+    /**
+     * If network is valid, do nothing.<br>
```

Comment:
How do we get a report in that case even if we have reach the minimum validation level?